### ISSUE TITLE
Typescript: Regex minify of GLSL files

### ISSUE THREAD

**Issue by wipfli** [maplibre/maplibre-gl-js#250]
We minify with home-made regular expressions in https://github.com/maplibre/maplibre-gl-js/blob/typescript/rollup/copy-glsl-files.js. Probably this should be done with a library.

**Comment by HarelM** [comment]
@JannikGM If you can point us to a library that worked well for you it will be great!

**Comment by JannikGM** [comment]
> @JannikGM If you can point us to a library that worked well for you it will be great!

For our companys shaders I tried a couple of different solutions in the past, but ended up using the GLSL processor that's also being used by Mozilla right now: https://github.com/jamienicol/glsl-optimizer/issues/5#issuecomment-842149817
However, glsl-optimizer needs proper GLSL so I have some custom code-generators and pre-processors written in python3. The output is standard GLSL, which is then post-processed by these scripts (generating more standard GLSL).
\- This step is for optimizing performance and working around WebGL driver parser and GLSL compiler issues.

As a final post-processing step I also wrote some python3 scripts to verify our code against assumed GLSL limits (number of `varying` / `uniform` / ...).
\- This is for avoiding bugs and crashed at runtime.

I don't do any additional minification after that, as our code is gzip compressed by the webserver anyway, so minification didn't seem to affect bundle size a lot (our webapp is multi-Megabyte sized anyway, so the shaders don't influence much).

I might eventually make some of the python3 code public, but currently I don't have time to refactor it / make it stand-alone. However, I don't think it's suitable for maplibre as-is anyway (being python3 and requiring a different GLSL setup than what's present in maplibre).
